src/components/QRShare/QRShareWithDownload/index.native.js

```diff
@@ -1,37 +1,46 @@
-import React, {Component} from 'react';
+import React, {forwardRef, useImperativeHandle, useRef} from 'react';
 import ViewShot from 'react-native-view-shot';
-import {withNetwork} from '@components/OnyxProvider';
 import getQrCodeFileName from '@components/QRShare/getQrCodeDownloadFileName';
 import {qrShareDefaultProps, qrSharePropTypes} from '@components/QRShare/propTypes';
+import useNetwork from '@hooks/useNetwork';
 import fileDownload from '@libs/fileDownload';
 import QRShare from '..';
 
-class QRShareWithDownload extends Component {
-    qrCodeScreenshotRef = React.createRef();
+function QRShareWithDownload({innerRef, ...props}) {
+    const {isOffline} = useNetwork();
+    const qrCodeScreenshotRef = useRef(null);
 
-    constructor(props) {
-        super(props);
+    useImperativeHandle(
+        innerRef,
+        () => ({
+            download: () => qrCodeScreenshotRef.current.capture().then((uri) => fileDownload(uri, getQrCodeFileName(props.title))),
+        }),
+        [props.title],
+    );
 
-        this.download = this.download.bind(this);
-    }
-
-    download() {
-        return this.qrCodeScreenshotRef.current.capture().then((uri) => fileDownload(uri, getQrCodeFileName(this.props.title)));
-    }
-
-    render() {
-        return (
-            <ViewShot ref={this.qrCodeScreenshotRef}>
-                <QRShare
-                    // eslint-disable-next-line react/jsx-props-no-spreading
-                    {...this.props}
-                    logo={this.props.network.isOffline ? null : this.props.logo}
-                />
-            </ViewShot>
-        );
-    }
+    return (
+        <ViewShot ref={qrCodeScreenshotRef}>
+            <QRShare
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...props}
+                logo={isOffline ? null : props.logo}
+            />
+        </ViewShot>
+    );
 }
+
 QRShareWithDownload.propTypes = qrSharePropTypes;
 QRShareWithDownload.defaultProps = qrShareDefaultProps;
+QRShareWithDownload.displayName = 'QRShareWithDownload';
+
+const QRShareWithDownloadWithRef = forwardRef((props, ref) => (
+    <QRShareWithDownload
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+        innerRef={ref}
+    />
+));
```

Comment:
You can directly use forwardRef in the top declaration only. Will keep render tree a level down